File: src/main/java/com/harragan/battleshipsboot/controllers/ShipPlaceRequestWrapper.java
package com.harragan.battleshipsboot.controllers;

import com.harragan.battleshipsboot.model.game.BoardPosition;
import com.harragan.battleshipsboot.model.game.Orientation;

public class ShipPlaceRequestWrapper {
  private int playerId;
  private int gameId;
  private BoardPosition boardPosition;
  private Orientation orientation;

  public ShipPlaceRequestWrapper() {}

  public ShipPlaceRequestWrapper(
      int playerId, int gameId, BoardPosition boardPosition, Orientation orientation) {
    this.playerId = playerId;
    this.gameId = gameId;
    this.boardPosition = boardPosition;
    this.orientation = orientation;
  }

  public int getPlayerId() {
    return playerId;
  }

  public int getGameId() {
    return gameId;
  }

  public BoardPosition getBoardPosition() {
    return boardPosition;
  }

  public Orientation getOrientation() {
    return orientation;
  }
}

File: src/main/java/com/harragan/battleshipsboot/controllers/ShipPlaceRequestWrapper.kt
package com.harragan.battleshipsboot.controllers

import com.harragan.battleshipsboot.model.kotlinmodel.game.BoardPosition
import com.harragan.battleshipsboot.model.kotlinmodel.game.Orientation
import com.harragan.battleshipsboot.model.kotlinmodel.ships.ShipType

data class ShipPlaceRequestWrapper(
        val playerId: Int,
        val gameId: Int,
        val boardPosition: BoardPosition,
        val orientation: Orientation,
        val shipType: ShipType
)